src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -140,46 +140,30 @@ private DirectoryInfo GetTestDestinationDirectory(string callingMethod, string i
 
         private IEnumerable<FileInfo> LoadInventory(FileInfo file)
         {
+            file.Refresh();
             if (!file.Exists)
             {
-                return Enumerable.Empty<FileInfo>();
+                throw new InvalidOperationException("Inventory file should exist.");
             }
 
             var inventory = new List<FileInfo>();
-
-            var lines = file.OpenText();
-
-            while (lines.Peek() > 0)
+            foreach (var p in File.ReadAllLines(file.FullName))
             {
-                inventory.Add(new FileInfo(lines.ReadLine()));
+                inventory.Add(new FileInfo(p));
             }
 
             return inventory;
         }
 
         private void SaveInventory(FileInfo file, IEnumerable<FileInfo> inventory)
         {
-            FileUtility.ReplaceWithLock(
-                filePath =>
-                {
-                    if (!_dataDirectory.Exists)
-                    {
-                        _dataDirectory.Create();
-                    }
+            _dataDirectory.Refresh();
```

Comment:
Is it a problem if two instances both try to create the directory? Will the two different instances try to create the same folder structure and same files? 

If they would both create the same output, then the correct answer is probably idempotency rather than pessimistic locking, e.g. swallow the exception and move on with the understanding that the folder was created concurrently but all is well.

On the other hand, if the outcomes are different, then I think the concurrency problem is actually a test design issue that has to be resolved higher up.